src/NuGet.Clients/NuGet.VisualStudio/Extensibility/IVsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// Contains methods to migrate a project.json based legacy project to PackageReference based project.
+    /// </summary>
+    [ComImport]
+    [Guid("2CB0AF9B-241D-4201-99ED-00F796C416BB")]
+    public interface IVsProjectJsonToPackageReferenceMigrator
```

Comment:
Implementation of this API can't be there for VS14 or need to make sure all the code specific to VS15 is guarded well. Also, if we provide these APIs for VS14 as well then what would happens when someone tries to use for VS14, noop?

I dont see any harm having an nuget interop package for VS15